webapp/tests/karma/ts/components/snackbar.component.spec.ts

```diff
@@ -0,0 +1,192 @@
+    const message = 'important message';
+    component.ngOnInit();
+    store.overrideSelector(Selectors.getSnackbarContent, { message, action: undefined });
+    store.refreshState();
+
+    expect(component.active).to.equal(true);
+    expect(getElement('#snackbar.active')).to.exist;
+    expect(getElement('#snackbar.active .snackbar-message').innerText).to.equal(message);
+    expect(getElement('#snackbar.active .snackbar-action')).to.not.exist;
+    expect(setSnackbarContent.callCount).to.equal(0);
+
+    tick(4500);
+    expect(component.active).to.equal(true);
+    expect(getElement('#snackbar.active')).to.exist;
+    expect(getElement('#snackbar.active .snackbar-message').innerText).to.equal(message);
+
+    tick(500);
+
+    expect(setSnackbarContent.callCount).to.equal(1);
+    expect(setSnackbarContent.firstCall.firstArg).to.be.undefined;
+    store.overrideSelector(Selectors.getSnackbarContent, { message: undefined, action: undefined });
+    store.refreshState();
+    expect(component.active).to.equal(false);
+    expect(getElement('#snackbar')).to.exist;
+    expect(getElement('#snackbar.active')).to.not.exist;
+  }));
+
+  it('should display the snackbar with a clickable action', async () => {
+    const message = 'important message';
+    const action = { label: 'click me', onClick: sinon.stub() };
+    component.ngOnInit();
+    store.overrideSelector(Selectors.getSnackbarContent, { message, action });
+    store.refreshState();
+
+    expect(component.active).to.equal(true);
+    expect(getElement('#snackbar.active')).to.exist;
+    expect(getElement('#snackbar.active .snackbar-message').innerText).to.equal(message);
+
+    const actionElement = getElement('#snackbar.active .snackbar-action');
+    expect(actionElement).to.exist;
+    expect(actionElement.innerText).to.equal(action.label);
+
+    expect(action.onClick.callCount).to.equal(0);
+    actionElement.click();
+    expect(action.onClick.callCount).to.equal(1);
+  });
+
+  it('should display the snackbar with a message and queue another one', fakeAsync(async () => {
+    const firstMessage = 'first message';
+    const secondMessage = 'second message';
+
+    component.ngOnInit();
+    store.overrideSelector(Selectors.getSnackbarContent, { message: firstMessage, action: undefined });
+    store.refreshState();
+    expect(getElement('#snackbar.active .snackbar-message').innerText).to.equal(firstMessage);
+
+    // second message gets queued
+    expect(setSnackbarContent.callCount).to.equal(0);
+    store.overrideSelector(Selectors.getSnackbarContent, { message: secondMessage, action: undefined });
+    store.refreshState();
+    expect(getElement('#snackbar.active .snackbar-message').innerText).to.equal(firstMessage);
+
+    tick(250);
+    // first message disappears
+    expect(setSnackbarContent.callCount).to.equal(1);
+    expect(setSnackbarContent.firstCall.args[0]).to.be.undefined;
```

Comment:
A shorter version of this is: 
```suggestion
    expect(setSnackbarContent.args[0][0]).to.deep.equal([]);
```